ui/src/app/machines/components/ActionFormWrapper/CloneForm/CloneForm.test.tsx

```diff
@@ -0,0 +1,120 @@
+import { mount } from "enzyme";
+import { Provider } from "react-redux";
+import { MemoryRouter, Route } from "react-router-dom";
+import configureStore from "redux-mock-store";
+
+import CloneForm from "./CloneForm";
+
+import { actions as machineActions } from "app/store/machine";
+import {
+  machine as machineFactory,
+  machineState as machineStateFactory,
+  machineStatus as machineStatusFactory,
+  machineStatuses as machineStatusesFactory,
+  rootState as rootStateFactory,
+} from "testing/factories";
+import { waitForComponentToPaint } from "testing/utils";
+
+const mockStore = configureStore();
+
+describe("CloneForm", () => {
+  it("can dispatch an action to clone to selected machines in the machine list", async () => {
+    const machines = [
+      machineFactory({ system_id: "abc123" }),
+      machineFactory({ system_id: "def456" }),
+      machineFactory({ system_id: "ghi789" }),
+    ];
+    const state = rootStateFactory({
+      machine: machineStateFactory({
+        active: null,
+        items: machines,
+        loaded: true,
+        selected: ["abc123", "def456"],
+        statuses: machineStatusesFactory({
+          abc123: machineStatusFactory(),
+          def456: machineStatusFactory(),
+          ghi789: machineStatusFactory(),
+        }),
+      }),
+    });
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <CloneForm clearSelectedAction={jest.fn()} />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    wrapper.find("Formik").invoke("onSubmit")({
+      interfaces: true,
+      source: "ghi789",
+      storage: false,
+    });
+    await waitForComponentToPaint(wrapper);
```

Comment:
Is this required when using `invoke` above? I thought that handled the async bits, but maybe it still has to wait.

Reply:
Oh yeah you're right, not sure why I included it!